app/src/main/java/de/grobox/transportr/settings/SettingsManager.kt

```diff
@@ -135,6 +136,30 @@ class SettingsManager @Inject constructor(private val context: Context) {
         return settings.getBoolean(SHOW_WHEN_LOCKED, true)
     }
 
+    fun setPreferredProducts(selected: Set<Product>) {
+        val editor = settings.edit()
+        Product.ALL.toSet().forEach { product ->
+            editor.putBoolean(product.name, product in selected)
```

Comment:
Maybe it would be actually better to use some common prefix for those setting entries, just to not pollute the setting file too much. Something like `lastProducts-*` maybe? Obviously this would need to be updated below in the other functions as well.

Reply:
Sounds good. I will update :+1: